guacamole/src/main/java/org/apache/guacamole/rest/APIRequest.java

```diff
@@ -58,6 +68,27 @@ public APIRequest(HttpServletRequest request,
 
         super(request);
 
+        // Try a few methods to get client info.
+        if(request.getHeader("X-Guacamole-Client-Hostname") != null && request.getHeader("X-Guacamole-Client-Hostname") != "") {
+            this.remoteHost = request.getHeader("X-Guacamole-Client-Hostname");
+        } else if(request.getHeader("X-Forwarded-For") != null && request.getHeader("X-Forwarded-For") != "") {
+            this.remoteHost = request.getHeader("X-Forwarded-For");
+        } else if(request.getRemoteHost() != null && request.getRemoteHost() != "") {
+            this.remoteHost = request.getRemoteHost();
+        } else {
+            this.remoteHost = "";
+        }
+
+        if(request.getHeader("X-Guacamole-Client-IP") != null && request.getHeader("X-Guacamole-Client-IP") != "") {
```

Comment:
Please follow the same style as the other code in guacamole-client for the sake of readability. There should be a space between the `if` and the `(`, as `if` is not a function, and we try to avoid:

* Using braces when unnecessary (unless it hurts readability)
* Cuddling the brace with the `else` or `else if`

If unclear, we also have [general style guidelines](http://guacamole.incubator.apache.org/guac-style/), but the best way is usually to simply take a quick look around and make sure your code doesn't stand out.